### ISSUE TITLE
UnicodeDecodeError with warnings for UTF-8 file paths

### ISSUE THREAD

**Issue by laeti-tia** [ansible/ansible#30592]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
CLI error/warnings reporting when running from a path containing UTF-8 chars

##### ANSIBLE VERSION
```
ansible 2.4.0.0
  config file = /Users/antoine/Jobs/GÉANT/Code/github/distribution/debian/build-machine/ansible.cfg
  configured module search path = [u'/Users/antoine/Jobs/GE\u0301ANT/Code/github/distribution/debian/build-machine/ansible/library', u'/Users/antoine/Library/Application Support/debops/debops-playbooks/playbooks/library', u'/usr/share/ansible/library']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.13 (default, Jul 18 2017, 09:17:00) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

##### CONFIGURATION
```
inventory = /Users/antoine/Jobs/GÉANT/Code/github/distribution/debian/build-machine/ansible/inventory
```

`ansible-config dump --only-changed` is actually giving a UnicodeDecodeError too.

##### OS / ENVIRONMENT
Running ansible on macos 10.12.6

##### SUMMARY
When you run ansible from a path containing UTF-8 chars and a warning or a coloured output needs to be printed by CLI tools, an UnicodeDecodeError happens.

##### STEPS TO REPRODUCE
With a simple playbook:
```yaml
---
- hosts: localhost
  tasks:
  - ping:
```

and a simple `ansible.cfg` file:
```yaml
[defaults]
inventory = /Users/antoine/café/inventory
action_plugins = "/Users/antoine/café/actions_plugins"
```

Steps to reproduce:
```
$ mkdir café
$ cd café
$ LC_ALL=fr_BE.UTF-8 ansible-playbook ping.yml
$ LC_ALL=fr_BE.UTF-8 ansible-config dump --only-changed
```

##### EXPECTED RESULTS
Correctly formatted [WARNING] line containing the path to files like:
```
 [WARNING]: Unable to parse /Users/antoine/café/hosts as an inventory source
```

##### ACTUAL RESULTS
```
ansible-playbook 2.4.0.0
  config file = /Users/antoine/.ansible.cfg
  configured module search path = [u'/Users/antoine/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.13 (default, Jul 18 2017, 09:17:00) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
Using /Users/antoine/.ansible.cfg as config file
Unable to parse /Users/antoine/café/hosts as an inventory source
ERROR! Unexpected Exception, this is probably a bug: 'ascii' codec can't decode byte 0xcc in position 47: ordinal not in range(128)
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/bin/ansible-playbook", line 106, in <module>
    exit_code = cli.run()
  File "/usr/local/lib/python2.7/site-packages/ansible/cli/playbook.py", line 104, in run
    loader, inventory, variable_manager = self._play_prereqs(self.options)
  File "/usr/local/lib/python2.7/site-packages/ansible/cli/__init__.py", line 784, in _play_prereqs
    inventory = InventoryManager(loader=loader, sources=options.inventory)
  File "/usr/local/lib/python2.7/site-packages/ansible/inventory/manager.py", line 144, in __init__
    self.parse_sources()
  File "/usr/local/lib/python2.7/site-packages/ansible/inventory/manager.py", line 205, in parse_sources
    parse = self.parse_source(source, cache=cache)
  File "/usr/local/lib/python2.7/site-packages/ansible/inventory/manager.py", line 290, in parse_source
    display.warning("Unable to parse %s as an inventory source" % to_native(source))
  File "/usr/local/lib/python2.7/site-packages/ansible/utils/display.py", line 221, in warning
    self.display(new_msg, color=C.COLOR_WARN, stderr=True)
  File "/usr/local/lib/python2.7/site-packages/ansible/utils/display.py", line 114, in display
    msg = stringc(msg, color)
  File "/usr/local/lib/python2.7/site-packages/ansible/utils/color.py", line 93, in stringc
    return u"\n".join([u"\033[%sm%s\033[0m" % (color_code, t) for t in text.split(u'\n')])
UnicodeDecodeError: 'ascii' codec can't decode byte 0xcc in position 47: ordinal not in range(128)

```

**Comment by samdoran** [comment]
cc @abadger @bcoca 

Looks like we need to properly encode/decode paths on the way in/out.

**Comment by ice3** [comment]
@mscherer Hello, the bug seems to be fixed since v2.4.2.0-1

**Comment by mscherer** [comment]
Closing, we did verify the bug was fixed during pycon.fr ansible sprint (and tested it was indeed broken in the first place). Please reopen if we were wrong.
